jsonrpc/eth.go

```diff
@@ -576,11 +576,10 @@ func (e *Eth) NewBlockFilter() (interface{}, rpcError) {
 // to notify when the state changes (logs). To check if the state
 // has changed, call eth_getFilterChanges.
 func (e *Eth) NewFilter(filter *LogFilter) (interface{}, rpcError) {
-	if filter.BlockHash != nil && (filter.FromBlock != nil || filter.ToBlock != nil) {
-		return rpcErrorResponse(invalidParamsErrorCode, "invalid argument 0: cannot specify both BlockHash and FromBlock/ToBlock, choose one or the other", nil)
-	}
 	id, err := e.storage.NewLogFilter(*filter)
-	if err != nil {
+	if errors.Is(err, ErrFilterInvalidPayload) {
+		return rpcErrorResponse(invalidParamsErrorCode, "invalid argument 0: cannot specify both BlockHash and FromBlock/ToBlock, choose one or the other", nil)
```

Comment:
@KonradIT I think that probably here you can do this now:
```go
return rpcErrorResponse(invalidParamsErrorCode, err.Error(), nil)
```
It should take the error message from the `ErrFilterInvalidPayload` in `pgstorage.go`